### ISSUE TITLE
Chest contents lost when chest is broken

### ISSUE THREAD

**Issue by bluelightning32** [SpongePowered/SpongeForge#2701]
**I am currently running**
- SpongeForge version: spongeforge-1.12.2-2768-7.1.6-RC3665.jar
- Forge version: forge-1.12.2-14.23.5.2822-universal.jar
- Java version: 1.8.0_201
- Operating System: Fedora 29 64bit

- Plugins/Mods:
```
[20:41:45] [Server thread/INFO] [minecraft/DedicatedServer]: Plugins (9): Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge, Inventory Sync, JdaLibraryLoader,
LuckPerms, UltimateChat, spark
```

- Modpack: DD&SS 4.9, which includes ImmersiveEngineering-0.12-89.jar

I tried removing the Auto Pickup mod (which comes with DD&SS). It doesn't make a difference for the repro.

**Issue Description**
Repro steps:
1. Place down a chest.
2. Put something in the chest.
3. Break the chest.
4. If Auto Pickup is installed, the chest will go directly to your inventory, but the chest contents will be list. If Auto Pickup is not installed, the chest item will drop to the ground, but the chest contents will not and will be lost.

This bug does not reproduce with just Forge.

**Comment by bluelightning32** [comment]
Does not reproduce in spongeforge-1.12.2-2768-7.1.6-RC3629.jar.

**Comment by Shybella** [comment]
Also encountering this on RC3665 from 3616.

This also occurs on many different containers, not just vanilla.

**Comment by geegehub** [comment]
This is happening for me as well. Vanilla chests are losing all contents while broken. I haven't tested anything else.

Running:
SpongeForge: 1.12.2-2768-7.1.6-RC3666
Forge: 1.12.2 - 14.23.5.2768

**Comment by gabizou** [comment]
Found the issue, will test the logic move to see if it breaks anything else (order of operations).

**Comment by bluelightning32** [comment]
Confirmed fixed in spongeforge-1.12.2-2768-7.1.6-RC3671.jar for vanilla chests. Immersive Engineering Wooden Storage Crates are still losing items. I'll file a separate bug for that.
